types/src/chain_id.rs

```diff
@@ -1,18 +1,52 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
+use anyhow::{Error, Result};
 use serde::{Deserialize, Serialize};
+use std::{
+    fmt::{Display, Formatter},
+    str::FromStr,
+};
 
-#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
-pub struct ChainId(pub String);
+#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct ChainId(pub u64);
+
+const MAINNET_CHAIN_ID: u64 = 0;
+#[cfg(any(test, feature = "fuzzing"))]
+pub const MOCK_CHAIN_ID: u64 = 1;
```

Comment:
should we just have TESTNET_CHAIN_ID here and for all tests? mock gives impression it is generated number. 
also, since 0 is mainnet, 1 wil be testnet?